### PR TITLE
Prepare Phpstan level 2

### PR DESCRIPTION
**Pull request by VincentLanglet**
We're accessing public property of ClassMetadataInfo, so this cannot be a ClassMetadataInterface.

### CODE REVIEW

**Review by phansys — APPROVED** [review]

**Review by core23 — APPROVED** [review]

File: src/Filter/Filter.php [hunk 1]
```diff
@@ -26,6 +26,10 @@ abstract class Filter extends BaseFilter
 
     /**
      * Apply the filter to the QueryBuilder instance.
+     *
+     * @param mixed[] $data
+     *
+     * @phpstan-param array{type?: string|int, value?: mixed} $data
```
Comment (core23) [inline src/Filter/Filter.php:32]: Can you port this to the stable branch?

File: src/Model/ModelManager.php [hunk 2]
```diff
@@ -403,13 +405,14 @@ public function addIdentifiersToQuery(string $class, ProxyQueryInterface $query,
 
     public function batchDelete(string $class, ProxyQueryInterface $query): void
     {
-        $query->select('DISTINCT '.current($query->getRootAliases()));
+        $qb = $query->getQueryBuilder();
```
Comment (core23) [inline src/Model/ModelManager.php:408]: Same for this

File: src/Filter/Filter.php (hunk 1 above)
Comment (VincentLanglet) [reply src/Filter/Filter.php:32]: This is already in the stable branch

File: src/Model/ModelManager.php (hunk 2 above)
Comment (VincentLanglet) [reply src/Model/ModelManager.php:408]: Check https://github.com/sonata-project/SonataDoctrineORMAdminBundle/pull/1326 :)

**Pull request merged**